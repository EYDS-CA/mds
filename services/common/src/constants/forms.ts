export enum FORM {
  // projects
  ADD_EDIT_PROJECT_SUMMARY = "ADD_EDIT_PROJECT_SUMMARY",
  ADD_MINE_MAJOR_APPLICATION = "ADD_MINE_MAJOR_APPLICATION",
  // reports
  VIEW_EDIT_REPORT = "VIEW_EDIT_REPORT",
  // permits
  REVOKE_DIGITAL_CREDENTIAL = "REVOKE_DIGITAL_CREDENTIAL",
  REQUEST_REPORT = "REQUEST_REPORT",
  ADD_REPORT_TO_PERMIT_CONDITION = "ADD_REPORT_TO_PERMIT_CONDITION",
  ADD_PERMIT_CONDITION_CATEGORY = "ADD_PERMIT_CONDITION_CATEGORY",
  EDIT_PERMIT_CONDITION = "EDIT_PERMIT_CONDITION",
  // tailings
  ADD_TAILINGS = "ADD_TAILINGS",
  // documents
  REPLACE_DOCUMENT = "REPLACE_DOCUMENT",
  DELETE_DOCUMENT = "DELETE_DOCUMENT",
  ARCHIVE_DOCUMENT = "ARCHIVE_DOCUMENT",
  EDIT_HELP_GUIDE = "EDIT_HELP_GUIDE",
  INLINE_EDIT_PERMIT_CONDITION_CATEGORY = "INLINE_EDIT_PERMIT_CONDITION_CATEGORY",
  PERMIT_CONDITION_REVIEW_ASSIGNMENT = "PERMIT_CONDITION_REVIEW_ASSIGNMENT",
  UPDATE_MAJOR_MINE_APPLICATION = "UPDATE_MAJOR_MINE_APPLICATION",
  UPDATE_IRT = "UPDATE_IRT"
}
