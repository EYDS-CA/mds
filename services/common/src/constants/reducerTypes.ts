// Misc
export const AUTHENTICATION = "AUTHENTICATION";
export const STATIC_CONTENT = "STATIC_CONTENT";
export const MODAL = "MODAL";
export const DOCUMENT_VIEWER = "DOCUMENT_VIEWER";
export const SYSTEM_FLAG = "SYSTEM_FLAG";

// Mines
export const MINES = "MINES";

// Parties
export const PARTIES = "PARTIES";

// OrgBook
export const ORGBOOK = "ORGBOOK";

// Compliance Orders
export const COMPLIANCE = "COMPLIANCE";

// MineSpace
export const MINESPACE = "MINESPACE";

// Permits
export const PERMITS = "PERMITS";

// Explosive Storage & Use Permits
export const EXPLOSIVES_PERMITS = "EXPLOSIVES_PERMITS";

// Search
export const SEARCH = "SEARCH";

// Variances
export const VARIANCES = "VARIANCES";

// Projects
export const PROJECTS = "PROJECTS";
export const PROJECT_SUMMARIES = "PROJECT_SUMMARIES";
export const INFORMATION_REQUIREMENTS_TABLE = "INFORMATION_REQUIREMENTS_TABLE";

// Core Users
export const USERS = "USERS";

// Incidents
export const INCIDENTS = "INCIDENTS";

// Work Information
export const MINE_WORK_INFORMATIONS = "MINE_WORK_INFORMATIONS";

// Reports
export const REPORTS = "REPORTS";

// Notices of Work
export const NOTICE_OF_WORK = "NOTICE_OF_WORK";

// Securities
export const SECURITIES = "SECURITIES";

// EPIC Information
export const EXTERNAL_AUTHS = "EXTERNAL_AUTHS";

// Notices of Departure
export const NOTICES_OF_DEPARTURE = "NOTICES_OF_DEPARTURE";

// Activities
export const ACTIVITIES = "ACTIVITIES";

// Tailings Storage Facilities
export const TAILINGS = "TAILINGS";

// Dams
export const DAMS = "DAMS";


//Verifiable Credentials
export const VERIFIABLE_CREDENTIALS = "VERIFIABLE_CREDENTIALS";
