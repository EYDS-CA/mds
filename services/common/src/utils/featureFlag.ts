import { detectProdEnvironment as IN_PROD } from "./environmentUtils";
import flagsmith from "flagsmith";

// Name of feature flags. These correspond to feature flags defined in flagsmith.
export enum Feature {
  MAJOR_PROJECT_ARCHIVE_FILE = "major_project_archive_file",
  DOCUMENTS_REPLACE_FILE = "major_project_replace_file",
  MAJOR_PROJECT_ALL_DOCUMENTS = "major_project_all_documents",
  MAJOR_PROJECT_DECISION_PACKAGE = "major_project_decision_package",
  FLAGSMITH = "flagsmith",
  TSF_V2 = "tsf_v2",
}

// Definition of legacy flags
// TODO: Remove after flagsmith is live
const Flags = {
  [Feature.MAJOR_PROJECT_ARCHIVE_FILE]: !IN_PROD(),
  [Feature.DOCUMENTS_REPLACE_FILE]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_ALL_DOCUMENTS]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_DECISION_PACKAGE]: !IN_PROD(),
  [Feature.TSF_V2]: !IN_PROD(),
  [Feature.FLAGSMITH]: !IN_PROD(),
};

export const initializeFlagsmith = async (flagsmithUrl, flagsmithKey) => {
  await flagsmith.init({
    api: flagsmithUrl,
    environmentID: flagsmithKey,
    cacheFlags: true,
    enableAnalytics: true,
  });
};

/**
 * Returns true if the given feature is enabled
 * @param feature Feature to verify
 * @returns true if the given feature is enabled
 */
export const isFeatureEnabled = (feature: Feature) => {
  if (Feature.FLAGSMITH in Flags && Flags[Feature.FLAGSMITH]) {
    return flagsmith.hasFeature(feature);
  }

  if (feature in Flags) {
    return Flags[feature];
  }

  return false;
};
