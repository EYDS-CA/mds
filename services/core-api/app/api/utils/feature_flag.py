from enum import Enum

from app.config import Config
from flagsmith import Flagsmith
from flask import current_app


class Feature(Enum):
    TSF_V2 = 'tsf_v2'
    MAJOR_PROJECT_REPLACE_FILE = 'major_project_replace_file'
    MINE_APPLICATION_FILE_UDPATE_ALERTS = 'mine_application_file_update_alerts'
    TRACTION_VERIFIABLE_CREDENTIALS = 'verifiable_credentials'
    #if enabled the credential offer will be the current development state of all the 2.0 changes, Q1 2024
    VC_ANONCREDS_20 = 'vc_mines_act_permit_20'   # pending anoncred content differences.
    VC_ANONCREDS_CORE = "vc_anoncreds_core"
    VC_ANONCREDS_MINESPACE = "vc_anoncreds_minespace"
    VC_W3C = "vc_w3c"
    CODE_REQUIRED_REPORTS = 'code_required-reports'
    PERMIT_DOCUMENT_KEYWORD_SEARCH = 'permit_document_keyword_search'
    AMS_AGENT = 'ams_agent'

    def __str__(self):
        return self.value


flagsmith = Flagsmith(
    environment_key=Config.FLAGSMITH_KEY,
    api_url=Config.FLAGSMITH_URL,
)


def is_feature_enabled(feature):
    try:
        feature = str(feature).strip()
        flags = flagsmith.get_environment_flags()

        return feature in flags.flags and flags.is_feature_enabled(feature)
    except Exception as e:
        current_app.logger.error(f'Failed to look up feature flag for: {feature}. ' + str(e))
        return False
