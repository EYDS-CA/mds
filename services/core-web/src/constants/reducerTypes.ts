// Documents
export const DOCUMENTS = "DOCUMENTS";

// Alerts
export const MINE_ALERTS = "MINE_ALERTS";