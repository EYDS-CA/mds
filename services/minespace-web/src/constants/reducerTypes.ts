// User Dashboard Info
export const USER_MINE_INFO = "USER_MINE_INFO";
