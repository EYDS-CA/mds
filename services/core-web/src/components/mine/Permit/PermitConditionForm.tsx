import React, { FC, useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { useParams } from "react-router-dom";
import { change, Field, reset } from "redux-form";
import { Row, Col, Button } from "antd";
import {
    faArrowDown,
    faArrowUp,
    faCheck,
    faClipboard,
    faPlus,
    faTrashCan,
    faXmark,
} from "@fortawesome/pro-regular-svg-icons";
import { FORM, IPermitCondition } from "@mds/common";
import { IGroupedDropdownList } from "@mds/common/interfaces/common/option.interface";
import { ERROR } from "@mds/common/constants/actionTypes";
import FormWrapper from "@mds/common/components/forms/FormWrapper";
import RenderAutoSizeField from "@mds/common/components/forms/RenderAutoSizeField";
import RenderCancelButton from "@mds/common/components/forms/RenderCancelButton";
import RenderSubmitButton from "@mds/common/components/forms/RenderSubmitButton";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { closeModal, openModal } from "@mds/common/redux/actions/modalActions";
import { ReportPermitRequirementForm } from "../../Forms/reports/ReportPermitRequirementForm";
import { deletePermitCondition, updatePermitCondition } from "@mds/common/redux/actionCreators/permitActionCreator";
import { createMineReportPermitRequirement } from "@mds/common/redux/slices/mineReportPermitRequirementSlice";
import RenderField from "@mds/common/components/forms/RenderField";
import { deleteConfirmWrapper } from "@mds/common/components/common/ActionMenu";
import { formatPermitConditionStep, parsePermitConditionStep } from "@mds/common/utils/helpers";
import RenderGroupedSelect from "@mds/common/components/forms/RenderGroupedSelect";


interface PermitConditionFormProps {
    permitAmendmentGuid: string;
    condition: IPermitCondition;
    canEditPermitConditions: boolean;
    onEdit: () => void;
    setEditingConditionGuid: (condition_guid: string) => void;
    editingConditionGuid: string;
    moveUp?: (condition: IPermitCondition) => Promise<void>;
    moveDown?: (condition: IPermitCondition) => Promise<void>;
    refreshData: () => Promise<void>;
    setIsAddingListItem: (isAdding: boolean) => void;
    isAddingListItem: boolean;
    categoryOptions?: IGroupedDropdownList[];
}
const PermitConditionForm: FC<PermitConditionFormProps> = ({
    permitAmendmentGuid,
    canEditPermitConditions,
    condition,
    onEdit,
    setEditingConditionGuid,
    editingConditionGuid,
    moveUp,
    moveDown,
    refreshData,
    setIsAddingListItem,
    isAddingListItem,
    categoryOptions
}) => {
    const dispatch = useDispatch();
    const { id: mineGuid, permitGuid } = useParams<{ id: string; permitGuid: string }>();
    const [isEditMode, setIsEditMode] = useState<boolean>(false);
    // the form fails to re-initialize when the category is changed, so concatenating it forces it to make a new one
    const formName = `${FORM.EDIT_PERMIT_CONDITION}_${condition.permit_condition_id}_${condition.condition_category_code}`;

    const startEdit = () => {
        onEdit();
        setEditingConditionGuid(condition.permit_condition_guid)
        setIsEditMode(true);
    };

    const cancelEdit = () => {
        setIsEditMode(false);
        setEditingConditionGuid(null);
        setIsAddingListItem(false);
    }

    // If the assigned user is changed while isEditMode
    // is true, set it to false
    useEffect(() => {
        if (!canEditPermitConditions) {
            setIsEditMode(false);
        }
    }, [canEditPermitConditions]);

    const handleSubmit = async (values) => {
        const payload = values.step
            ? {
                ...values,
                // Backend has the property named as _step to update in the db
                _step: values.step
            } : values;
        const resp = await dispatch(updatePermitCondition(values.permit_condition_guid, permitAmendmentGuid, payload));
        // @ts-ignore
        if (resp?.type !== ERROR) {
            cancelEdit();
            return refreshData();
        }
    };
    const handleCancel = () => {
        cancelEdit();
        dispatch(reset(formName));
    };
    const handleAddListItem = () => {
        setIsAddingListItem(true);
    };

    const handleDelete = async () => {
        deleteConfirmWrapper("Permit Condition", async () => {
            const resp = await dispatch(deletePermitCondition(permitAmendmentGuid, condition.permit_condition_guid));
            // @ts-ignore
            if (resp?.type !== ERROR) {
                refreshData();
                cancelEdit();
            }
        });
    };

    const addNewReport = async (values) => {
        await dispatch(createMineReportPermitRequirement({ mineGuid, values }));
        refreshData();
        dispatch(closeModal());
    };

    const handleOpenAddReportModal = (event, reportCondition: IPermitCondition) => {
        event.stopPropagation();
        dispatch(
            openModal({
                props: {
                    onSubmit: addNewReport,
                    title: `Add Permit Required Report to Condition`,
                    condition: reportCondition,
                    permitGuid,
                },
                content: ReportPermitRequirementForm,
            })
        );
    };

    const editableProps = !editingConditionGuid && canEditPermitConditions ?
        {
            onClick: startEdit,
            title: "Edit Condition",
            "aria-label": "Edit Condition",
        } : {};

    // deals with how the formatting prevents backspace
    const handleBackSpace = (event, value: string, prev: string, name: string) => {
        const { nativeEvent } = event;
        if (nativeEvent?.inputType === "deleteContentBackward" && value === prev) {
            event.preventDefault();
            const newVal = value.substring(0, value.length - 1);
            dispatch(change(formName, name, newVal))
        }
    };

    return (
        <FormWrapper
            isEditMode={isEditMode}
            onSubmit={handleSubmit}
            name={formName}
            initialValues={condition}
            scrollOnToggleEdit={false}
            reduxFormConfig={{
                enableReinitialize: true
            }}
        >
            {(isEditMode && categoryOptions) && <Row>
                <Col span={24}>
                    <Field
                        showOptional={false}
                        label="Condition Category:"
                        component={RenderGroupedSelect}
                        name="condition_category_code"
                        data={categoryOptions}
                        allowClear={false}
                        className="horizontal-form-item"
                    />
                </Col>
            </Row>}
            <Row
                wrap={false}
                align="top"
                className={`condition-content ${!editingConditionGuid ? "editable" : ""}`}
            >
                <Col className="step-column" style={{ flexShrink: 0 }}>
                    <Field
                        format={(value: string) => formatPermitConditionStep(value)}
                        parse={(value: string) => parsePermitConditionStep(value)}
                        name="step"
                        component={RenderField}
                        showNA={false}
                        disabled={isAddingListItem}
                        onChange={handleBackSpace}
                    />
                </Col>
                <Col className="condition-column"
                    {...editableProps}
                >
                    <Field
                        name="condition"
                        component={RenderAutoSizeField}
                        disabled={isAddingListItem}
                    />
                </Col>
            </Row>
            {isEditMode && !isAddingListItem && (
                <Row justify="space-between" align="middle">
                    <Col>
                        <Row gutter={8}
                            className="condition-edit-buttons"
                        >
                            <Col>
                                <Button
                                    className="fa-icon-container btn-sm-padding"
                                    type="default"
                                    icon={<FontAwesomeIcon icon={faPlus} />}
                                    onClick={handleAddListItem}
                                >List Item</Button>
                            </Col>
                            {/* <Col>
                                    <Button
                                        className="fa-icon-container btn-sm-padding"
                                        type="default"
                                        icon={<FontAwesomeIcon icon={faLink} />}
                                        onClick={handleLinkDocument}
                                    >
                                        Link Document
                                    </Button>
                                </Col> */}
                            <Col>
                                <Button
                                    className="fa-icon-container btn-sm-padding"
                                    type="default"
                                    icon={<FontAwesomeIcon icon={faClipboard} />}
                                    onClick={(e) => handleOpenAddReportModal(e, condition)}
                                >
                                    Add Report Requirement
                                </Button>
                            </Col>
                            <Col>
                                <RenderCancelButton
                                    cancelFunction={handleCancel}
                                    buttonProps={{
                                        type: "primary",
                                        icon: <FontAwesomeIcon icon={faXmark} />,
                                    }}
                                    iconButton

                                />
                            </Col>
                            <Col>
                                <RenderSubmitButton
                                    buttonProps={{
                                        icon: <FontAwesomeIcon icon={faCheck} />,
                                    }}
                                    iconButton
                                />
                            </Col>
                        </Row>
                    </Col>
                    <Col>
                        <Row gutter={8} align="middle" className="condition-edit-buttons">
                            <Col>
                                <Button
                                    className="fa-icon-container"
                                    aria-label="Delete Condition"
                                    type="default"
                                    icon={<FontAwesomeIcon icon={faTrashCan} />}
                                    onClick={handleDelete}
                                />
                            </Col>
                            <Col>
                                <Button
                                    className="fa-icon-container"
                                    aria-label="Move Condition Up"
                                    type="default"
                                    disabled={!moveUp}
                                    icon={<FontAwesomeIcon icon={faArrowUp} />}
                                    onClick={() => moveUp(condition)}
                                />
                            </Col>
                            <Col>
                                <Button
                                    className="fa-icon-container"
                                    aria-label="Move Condition Down"
                                    type="default"
                                    disabled={!moveDown}
                                    icon={<FontAwesomeIcon icon={faArrowDown} />}
                                    onClick={() => moveDown(condition)}
                                />
                            </Col>
                        </Row>
                    </Col>
                </Row>
            )}
        </FormWrapper>
    );
};

export default PermitConditionForm;