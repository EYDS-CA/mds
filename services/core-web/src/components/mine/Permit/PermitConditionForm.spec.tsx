import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";
import { PERMITS } from "@mds/common/constants/reducerTypes";
import * as MOCK from "@mds/common/tests/mocks/dataMocks";
import PermitConditionForm from "./PermitConditionForm";

const condition = MOCK.PERMITS[0].permit_amendments[0].conditions[0];
const initialState = {
    [PERMITS]: {
        permits: MOCK.PERMITS,
        permitAmendments: {
            [MOCK.PERMITS[0].permit_guid]: MOCK.PERMITS[0].permit_amendments[0]
        }
    },
};

function mockFunction() {
    const original = jest.requireActual("react-router-dom");
    return {
        ...original,
        useParams: jest.fn().mockReturnValue({
            id: MOCK.MINES.mineIds[0],
            permitGuid: MOCK.PERMITS[0].permit_guid,
        }),
    };
}

jest.mock("react-router-dom", () => mockFunction());

describe("PermitConditionForm", () => {
    it("renders properly in edit mode", async () => {
        const { container, findByText } = render(<ReduxWrapper initialState={initialState}>
            <PermitConditionForm
                permitAmendmentGuid={MOCK.PERMITS[0].permit_amendments[0].permit_amendment_guid}
                condition={condition}
                canEditPermitConditions={true}
                onEdit={jest.fn()}
                setEditingConditionGuid={jest.fn()}
                editingConditionGuid={undefined}
                moveUp={jest.fn()}
                moveDown={jest.fn()}
                refreshData={jest.fn()}
                setIsAddingListItem={jest.fn()}
                isAddingListItem={false}
            />
        </ReduxWrapper>);

        const editElement = container.querySelector("[aria-label='Edit Condition']")
        fireEvent.click(editElement);
        await findByText("Add Report Requirement");
        expect(container).toMatchSnapshot()
    });
    it("does not allow edit when it shouldn't", async () => {
        const { container } = render(<ReduxWrapper initialState={initialState}>
            <PermitConditionForm
                permitAmendmentGuid={MOCK.PERMITS[0].permit_amendments[0].permit_amendment_guid}
                condition={condition}
                canEditPermitConditions={false}
                onEdit={jest.fn()}
                setEditingConditionGuid={jest.fn()}
                editingConditionGuid={undefined}
                moveUp={jest.fn()}
                moveDown={jest.fn()}
                refreshData={jest.fn()}
                setIsAddingListItem={jest.fn()}
                isAddingListItem={false}
            />
        </ReduxWrapper>);

        const editElements = container.querySelectorAll("[aria-label='Edit Condition']");
        expect(Array.from(editElements)).toEqual([]);

        // and make sure that it's not just missing the aria-label and nothing happens on click
        const conditionColumn = container.querySelector(".condition-column");
        fireEvent.click(conditionColumn);
        await waitFor(() => {
            const editButtons = container.querySelectorAll(".condition-edit-buttons");
            expect(Array.from(editButtons)).toEqual([])
        })
    });
});