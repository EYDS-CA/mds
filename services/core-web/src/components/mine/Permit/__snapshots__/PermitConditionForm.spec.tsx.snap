// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermitConditionForm renders properly in edit mode 1`] = `
<div>
  <form
    class="ant-form ant-form-vertical common-form common-form-EDIT_PERMIT_CONDITION_1639_HSC form-edit"
    id="EDIT_PERMIT_CONDITION_1639_HSC"
  >
    <div
      class="ant-row ant-row-no-wrap ant-row-top condition-content editable"
    >
      <div
        class="ant-col step-column"
        style="flex-shrink: 0;"
      >
        <div
          class="ant-form-item ant-form-item-with-help"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    aria-label="step"
                    class="ant-input"
                    name="step"
                    type="text"
                    value="1."
                  />
                </div>
              </div>
              <div
                style="display: flex; flex-wrap: nowrap;"
              >
                <div
                  class="ant-form-item-explain ant-form-item-explain-connected"
                  id="EDIT_PERMIT_CONDITION_1639_HSC_step_help"
                  role="alert"
                >
                  <div
                    class=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="Edit Condition"
        class="ant-col condition-column"
        title="Edit Condition"
      >
        <div
          class="ant-form-item ant-form-item-with-help"
        >
          <div
            class="ant-row ant-form-item-row"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <textarea
                    class="ant-input"
                    name="condition"
                    style="height: 0px; resize: none; min-height: -12px;"
                  >
                    Section decade check including.
                  </textarea>
                </div>
              </div>
              <div
                style="display: flex; flex-wrap: nowrap;"
              >
                <div
                  class="ant-form-item-explain ant-form-item-explain-connected"
                  id="EDIT_PERMIT_CONDITION_1639_HSC_condition_help"
                  role="alert"
                >
                  <div
                    class=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-space-between ant-row-middle"
    >
      <div
        class="ant-col"
      >
        <div
          class="ant-row condition-edit-buttons"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <button
              class="ant-btn ant-btn-default fa-icon-container btn-sm-padding"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-plus "
                data-icon="plus"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M248 72c0-13.3-10.7-24-24-24s-24 10.7-24 24V232H40c-13.3 0-24 10.7-24 24s10.7 24 24 24H200V440c0 13.3 10.7 24 24 24s24-10.7 24-24V280H408c13.3 0 24-10.7 24-24s-10.7-24-24-24H248V72z"
                  fill="currentColor"
                />
              </svg>
              <span>
                List Item
              </span>
            </button>
          </div>
          <div
            class="ant-col"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <button
              class="ant-btn ant-btn-default fa-icon-container btn-sm-padding"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-clipboard "
                data-icon="clipboard"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M280 64h40c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128C0 92.7 28.7 64 64 64h40 9.6C121 27.5 153.3 0 192 0s71 27.5 78.4 64H280zM64 112c-8.8 0-16 7.2-16 16V448c0 8.8 7.2 16 16 16H320c8.8 0 16-7.2 16-16V128c0-8.8-7.2-16-16-16H304v24c0 13.3-10.7 24-24 24H192 104c-13.3 0-24-10.7-24-24V112H64zm128-8a24 24 0 1 0 0-48 24 24 0 1 0 0 48z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Add Report Requirement
              </span>
            </button>
          </div>
          <div
            class="ant-col"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <button
              aria-label="Cancel"
              class="ant-btn ant-btn-primary ant-btn-icon-only core-btn core-btn-primary"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-xmark "
                data-icon="xmark"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M345 137c9.4-9.4 9.4-24.6 0-33.9s-24.6-9.4-33.9 0l-119 119L73 103c-9.4-9.4-24.6-9.4-33.9 0s-9.4 24.6 0 33.9l119 119L39 375c-9.4 9.4-9.4 24.6 0 33.9s24.6 9.4 33.9 0l119-119L311 409c9.4 9.4 24.6 9.4 33.9 0s9.4-24.6 0-33.9l-119-119L345 137z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
          <div
            class="ant-col"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <button
              aria-label="Submit"
              class="ant-btn ant-btn-primary ant-btn-icon-only"
              disabled=""
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check "
                data-icon="check"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M441 103c9.4 9.4 9.4 24.6 0 33.9L177 401c-9.4 9.4-24.6 9.4-33.9 0L7 265c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l119 119L407 103c9.4-9.4 24.6-9.4 33.9 0z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="ant-col"
      >
        <div
          class="ant-row ant-row-middle condition-edit-buttons"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <button
              aria-label="Delete Condition"
              class="ant-btn ant-btn-default ant-btn-icon-only fa-icon-container"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-trash-can "
                data-icon="trash-can"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M170.5 51.6L151.5 80h145l-19-28.4c-1.5-2.2-4-3.6-6.7-3.6H177.1c-2.7 0-5.2 1.3-6.7 3.6zm147-26.6L354.2 80H368h48 8c13.3 0 24 10.7 24 24s-10.7 24-24 24h-8V432c0 44.2-35.8 80-80 80H112c-44.2 0-80-35.8-80-80V128H24c-13.3 0-24-10.7-24-24S10.7 80 24 80h8H80 93.8l36.7-55.1C140.9 9.4 158.4 0 177.1 0h93.7c18.7 0 36.2 9.4 46.6 24.9zM80 128V432c0 17.7 14.3 32 32 32H336c17.7 0 32-14.3 32-32V128H80zm80 64V400c0 8.8-7.2 16-16 16s-16-7.2-16-16V192c0-8.8 7.2-16 16-16s16 7.2 16 16zm80 0V400c0 8.8-7.2 16-16 16s-16-7.2-16-16V192c0-8.8 7.2-16 16-16s16 7.2 16 16zm80 0V400c0 8.8-7.2 16-16 16s-16-7.2-16-16V192c0-8.8 7.2-16 16-16s16 7.2 16 16z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
          <div
            class="ant-col"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <button
              aria-label="Move Condition Up"
              class="ant-btn ant-btn-default ant-btn-icon-only fa-icon-container"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-up "
                data-icon="arrow-up"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M209.4 39.4C204.8 34.7 198.6 32 192 32s-12.8 2.7-17.4 7.4l-168 176c-9.2 9.6-8.8 24.8 .8 33.9s24.8 8.8 33.9-.8L168 115.9V456c0 13.3 10.7 24 24 24s24-10.7 24-24V115.9L342.6 248.6c9.2 9.6 24.3 9.9 33.9 .8s9.9-24.3 .8-33.9l-168-176z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
          <div
            class="ant-col"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <button
              aria-label="Move Condition Down"
              class="ant-btn ant-btn-default ant-btn-icon-only fa-icon-container"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-arrow-down "
                data-icon="arrow-down"
                data-prefix="far"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M174.6 472.6c4.5 4.7 10.8 7.4 17.4 7.4s12.8-2.7 17.4-7.4l168-176c9.2-9.6 8.8-24.8-.8-33.9s-24.8-8.8-33.9 .8L216 396.1 216 56c0-13.3-10.7-24-24-24s-24 10.7-24 24l0 340.1L41.4 263.4c-9.2-9.6-24.3-9.9-33.9-.8s-9.9 24.3-.8 33.9l168 176z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
